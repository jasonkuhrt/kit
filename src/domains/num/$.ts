/**
 * Numeric types and utilities with branded types for mathematical constraints.
 *
 * Provides branded number types (Positive, Negative, Even, Odd, etc.) with
 * runtime validation, mathematical operations, range types, and specialized
 * numeric domains like Complex, Ratio, and BigInt. Includes type guards,
 * ordering, and equivalence utilities.
 */
export * as Num from './$$.js'
