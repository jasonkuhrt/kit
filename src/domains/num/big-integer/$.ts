export * as BigInteger from './$$.ts'
