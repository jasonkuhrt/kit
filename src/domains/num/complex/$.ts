export * as Complex from './$$.js'
