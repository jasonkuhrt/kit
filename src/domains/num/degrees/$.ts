// Export the namespace containing all Degrees operations
export * as Degrees from './$$.js'
