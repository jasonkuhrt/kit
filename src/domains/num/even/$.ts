// Export the namespace containing all Even operations
export * as Even from './$$.ts'
