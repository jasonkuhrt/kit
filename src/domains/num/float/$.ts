// Export the namespace containing all Float operations
export * as Float from './$$.js'
