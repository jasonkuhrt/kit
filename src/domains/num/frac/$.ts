// Export the namespace containing all Frac operations
export * as Frac from './$$.js'
