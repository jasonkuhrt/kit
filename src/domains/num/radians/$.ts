// Export the namespace containing all Radians operations
export * as Radians from './$$.js'
