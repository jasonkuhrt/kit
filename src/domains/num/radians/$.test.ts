import { Test } from '#test'
import { Radians } from './$.js'

Test.describe('is')
  .on(Radians.is)
  .casesInput(
    // Valid cases - any finite number is a valid radian
    [0],
    [Math.PI],
    [Math.PI / 2],
    [2 * Math.PI],
    [-Math.PI],
    [1.5708],
    // Invalid cases - non-finite numbers
    [Infinity],
    [-Infinity],
    [NaN],
    ['3.14'],
    [null],
    [undefined],
  )
  .test()

Test.describe('from')
  .on(Radians.from)
  .cases(
    // Valid cases
    [0],
    [Math.PI],
    [Math.PI / 2],
    [2 * Math.PI],
    [-Math.PI],
    [1.5708],
    // Invalid cases - should throw
    [Infinity],
    [-Infinity],
    [NaN],
  )
  .test()

Test.describe('tryFrom')
  .on(Radians.tryFrom)
  .cases(
    // Valid cases
    [0],
    [Math.PI],
    [Math.PI / 2],
    [2 * Math.PI],
    [-Math.PI],
    [1.5708],
    // Invalid cases - should return null
    [Infinity],
    [-Infinity],
    [NaN],
  )
  .test()
