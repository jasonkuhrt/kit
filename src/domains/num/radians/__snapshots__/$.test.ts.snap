// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`from > from(-3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-3.141592653589793
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1.5708) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5708
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1.5708
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(1.5707963267948966) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5707963267948966
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1.5707963267948966
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
3.141592653589793
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(6.283185307179586) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
6.283185307179586
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
6.283185307179586
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be finite, got: Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be finite, got: -Infinity
╚══════════════════════════════════════════════════╝"
`;

exports[`from > from(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN THROWS ERROR
Error: Value must be finite, got: NaN
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is("3.14") 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.14
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(-3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.5708) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5708
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(1.5707963267948966) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5707963267948966
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(6.283185307179586) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
6.283185307179586
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
true
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(null) 4`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`is > is(undefined) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
undefined
╠══════════════════════════════════════════════════╣ THEN RETURNS BOOLEAN
false
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(-3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
-3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
-3.141592653589793
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(0) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
0
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
0
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.5708) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5708
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1.5708
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(1.5707963267948966) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
1.5707963267948966
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
1.5707963267948966
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(3.141592653589793) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
3.141592653589793
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
3.141592653589793
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(6.283185307179586) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
6.283185307179586
╠══════════════════════════════════════════════════╣ THEN RETURNS NUMBER
6.283185307179586
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 1`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 2`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;

exports[`tryFrom > tryFrom(null) 3`] = `
"
╔══════════════════════════════════════════════════╗ GIVEN ARGUMENTS
null
╠══════════════════════════════════════════════════╣ THEN RETURNS NULL
null
╚══════════════════════════════════════════════════╝"
`;
